packages/extensions/openapi-to-typespec/test/arm-agrifood/tsp-output/DataConnector.tsp

```diff
@@ -20,25 +19,21 @@ namespace Azure.ResourceManager.AgFoodPlatform;
 @parentResource(DataManagerForAgriculture)
 @includeInapplicableMetadataInPayload(false)
 @Azure.ResourceManager.Private.armResourceInternal(DataConnectorProperties)
-model DataConnector extends ProxyResourceBase {
-  /**
-   * Connector name.
-   */
-  @maxLength(63)
-  @minLength(1)
-  @path
-  @key("dataConnectorName")
-  @segment("dataConnectors")
-  @visibility("read")
-  name: string;
+model DataConnector extends Foundations.ProxyResource {
+  ...ResourceNameParameter<
+    Resource = DataConnector,
+    KeyName = "dataConnectorName",
+    SegmentName = "dataConnectors",
+    NamePattern = ""
```

Comment:
will this empty still work for `@patten`?

Reply:
Yes, `ResourceNameParameter` has default pattern. If we don't pass empty to it, it will take the default one, which is not expected.